docs/src/lints/bad_string_escape.md

```diff
@@ -0,0 +1,26 @@
+# bad_string_escape
+## What it does
+Checks for invalid, malformed, or unnecessary string escape sequences.
+
+## Why this is bad
+Invalid string escapes don't do anything, so should obviously be caught in dealt with. Additionally, in double strings, you shouldn't escape single quote strings since it makes the string less readable. Same with single quote strings and double quotes.
+
+In some cases (specifically `\x` and `\u` when built with the `roblox` feature) it's possible to write an escape sequence that looks right but doesn't work when ran. Because this is probably not intentional, they are caught by this lint.
```

Comment:
User facing docs don't need to mention internals like the roblox feature flag. This should be changed to something more general, i.e. "in a Roblox codebase".